services-core/src/gas_price.rs

```diff
@@ -15,11 +16,17 @@ pub trait GasPriceEstimating {
 }
 
 /// Creates the default gas price estimator for the given network.
-pub fn create_estimator(
-    network_id: u64,
+pub async fn create_estimator(
     http_factory: &HttpFactory,
     web3: &Web3,
 ) -> Result<Arc<dyn GasPriceEstimating + Send + Sync>> {
+    let network_id: u64 = web3
+        .net()
+        .version()
+        .compat()
+        .await?
+        .parse()
```

Comment:
Ubernit: I think it's slightly more correct to use strings for network IDs, since according to the Ethereum JSON RPC spec, it is a string and not a number. While in practice this makes no difference, it may if some future chain we connect to doesn't return a number.